### DISCUSSION TITLE
Compiling a library at installation instead of precompiling?

### DISCUSSION THREAD

**Discussion by asterized** [PyO3/pyo3#3630] [Questions] [answered]
Is it possible to compile a library when installing a package instead of precompiling it and bundling?

**Comment by davidhewitt** [comment, marked as answer]
Sure, both maturin and setuptools-rust are designed to support this automatically on platforms without precompiled wheels by building the source distribution.

To get the result you want just don't distribute wheels, or alternatively opt out of downloading binary wheels when doing the pip install.

**Reply by asterized** [reply]
How would you do that?

\+ is it possible to add compiler flags on compilation?

**Reply by davidhewitt** [reply]
Follow the `setuptools-rust` or `maturin` documentation to package your project using them, and when you upload, only upload the `sdist`.

**Reply by asterized** [reply]
ty!
